### PR TITLE
Add security_group_names paramter to ec2_elb_lb

### LINKED ISSUE #1270: Standardise on using security group name instead of id
**Issue Type:**

Bug Report

**Ansible Version:**
1.9.1

**Environment:**
aws cloud modules

**Summary:**

In general I find the mix of using security_group_id some places, and the security group name other places  in aws modules cumbersome. 

ec2_elb_lb need to have the security group specified by group id, while ec2 module is happy with only the group name. 

Problem is that if the security group is already created using ec2_group it fails on the second run, and I can't register the variable and get hold of the security group id. So I either have to delete and recreate the security group, or hard-code the group id in my run.

**Steps To Reproduce:**
Run the following twice - (substitute my variables with meaningful vals for you):

``` yaml
- name: Create the security group for the ELBs
  local_action:
    module: ec2_group
    name: "{{ elb_public_vpc_group.name }}"
    description: "{{ elb_public_vpc_group.desc }}"
    vpc_id: "{{ vpc.vpc_id }}"
    region: "{{ ec2_region }}"
    ec2_access_key: "{{ ec2_access_key }}"
    ec2_secret_key: "{{ ec2_secret_key }}"
    rules: "{{ elb_public_vpc_group.rules }}"
    rules_egress: "{{ elb_public_vpc_group.rules_egress }}"
  register: elb_pub_group

- debug: var=elb_pub_group
```

**Expected Results:**

```
TASK: [vpc | debug var=elb_pub_group] *****************************************
ok: [localhost] => {
    "var": {
        "elb_pub_group": {
            "changed": true,
            "group_id": "sg-95e2baf0",
            "invocation": {
                "module_args": "",
                "module_name": "ec2_group"
            }
        }
    }
}
```

I would expect the module to query or update the group so that the current values can be used later on in the script even if the group is already created.

**Actual Results:**

```
TASK: [vpc | Create the security group for the ELBs] **************************
failed: [localhost -> 127.0.0.1] => {"failed": true, "parsed": false}
Traceback (most recent call last):
  File "/Users/olvesh/.ansible/tmp/ansible-tmp-1430921963.5-88774226046590/ec2_group", line 2192, in <module>
    main()
  File "/Users/olvesh/.ansible/tmp/ansible-tmp-1430921963.5-88774226046590/ec2_group", line 348, in main
    cidr_ip=ip)
  File "/usr/local/lib/python2.7/site-packages/boto/ec2/connection.py", line 3229, in authorize_security_group_egress
    params, verb='POST')
  File "/usr/local/lib/python2.7/site-packages/boto/connection.py", line 1226, in get_status
    raise self.ResponseError(response.status, response.reason, body)
boto.exception.EC2ResponseError: EC2ResponseError: 400 Bad Request
<?xml version="1.0" encoding="UTF-8"?>
<Response><Errors><Error><Code>InvalidPermission.Duplicate</Code><Message>the specified rule "peer: 0.0.0.0/0, ICMP, type: ALL, code: ALL, ALLOW" already exists</Message></Error></Errors><RequestID>efd596d8-219d-4323-9024-c283b9b55f0f</RequestID></Response>
```

**EDIT**
I removed the ICMP egress from the following ruleset and it worked for me, but still feel this is a bug.

``` yaml
elb_public_vpc_group:
  name: "{{ deploy_env }}-public"
  desc: the security group for the dev vpc
  rules:
    - proto: tcp
      from_port: 80
      to_port: 80
      cidr_ip: 0.0.0.0/0
    - proto: tcp
      from_port: 443
      to_port: 443
      cidr_ip: 0.0.0.0/0

  rules_egress:
    - proto: udp
      from_port: 0
      to_port: 65535
      cidr_ip: 0.0.0.0/0
    - proto: tcp
      from_port: 0
      to_port: 65535
      cidr_ip: 0.0.0.0/0
    - proto: icmp
      from_port: -1
      to_port: -1
      cidr_ip: 10.0.0.0/0
```

### PR DESCRIPTION
**Pull request by haad**
Make sure that ec2_elb_lb module is compatible with others and user can specify applicable security group with name not only id. 

This fixes #1270

### CODE REVIEW

File: cloud/amazon/ec2_elb_lb.py [hunk 1]
```diff
@@ -56,6 +56,11 @@
     require: false
     default: None
     version_added: "1.6"
+  security_group_names:
```
Comment (bcoca) [inline cloud/amazon/ec2_elb_lb.py:59]: ~~needs version_added~~ done

**Comment by haad** [comment]
This fixes #1270

**Comment by gregdek** [comment]
@haad Thanks for submitting this pull request. Please accept our apologies for taking so long to review it.

Unfortunately, due to this delay, it appears that the underlying code has changed, and this PR needs to be rebased. If you believe that this PR still applies, please rebase and we will then put it into our (new, faster) community review process as soon as possible.  Again, apologies for the delay and the extra work.

**Comment by haad** [comment]
Ok I have rebased it. Can we have a look at it again ?

**Comment by bcoca** [comment]
cc @defionscode can i get a +1?

File: cloud/amazon/ec2_elb_lb.py [hunk 2]
```diff
@@ -858,6 +868,23 @@ def main():
     if state == 'present' and not (zones or subnets):
         module.fail_json(msg="At least one availability zone or subnet is required for ELB creation")
 
+    if security_group_ids and security_group_names:
```
Comment (juliedavila) [inline cloud/amazon/ec2_elb_lb.py]: Would be preferred if `mutually exclusive` is leveraged within the argument spec object.

**Comment by juliedavila** [comment]
I'll +1, but would really prefer the update I mentioned for line 871

**Comment by haad** [comment]
@defionscode  here you go :)

**Comment by bcoca** [comment]
@haad I'm going to assume you just wanted the feature included and rebased from the original author (@rickmendes), but keep in mind that you can do that while keeping the authorship in the commits and you should mention that is what you are doing when opening a new PR.

**Pull request merged**